src/libs/DebugUtils.ts

```diff
@@ -527,6 +628,181 @@ function validateReportActionJSON(json: string) {
+            const iouReport = ReportUtils.getReportOrDraftReport(action.childReportID ?? '-1');
+            const policy = PolicyUtils.getPolicy(iouReport?.policyID);
+            const shouldShowSettlementButton = IOU.canIOUBePaid(iouReport, report, policy) || IOU.canApproveIOU(iouReport, report, policy);
+            return action.actionName === CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW && shouldShowSettlementButton;
+        });
+    }
+
+    return undefined;
+}
+
+/**
+ * Gets the report action that is causing the RBR to show up in LHN
+ *
+ * Based on OptionListUtils.getAllReportErrors
+ *
+ * @param report
+ * @param reportActions
+ * @returns report action
+ */
+function getRBRReportAction(report: OnyxEntry<Report>, reportActions: OnyxEntry<ReportActions>): OnyxEntry<ReportAction> {
+    const parentReportAction: OnyxEntry<ReportAction> =
+        !report?.parentReportID || !report?.parentReportActionID ? undefined : allReportActions?.[report.parentReportID ?? '-1']?.[report.parentReportActionID ?? '-1'];
+
+    // TODO: This branch is never reached because parentReportAction is undefined
+    if (ReportActionsUtils.wasActionTakenByCurrentUser(parentReportAction) && ReportActionsUtils.isTransactionThread(parentReportAction)) {
+        const transactionID = ReportActionsUtils.isMoneyRequestAction(parentReportAction) ? ReportActionsUtils.getOriginalMessage(parentReportAction)?.IOUTransactionID : null;
+        const transaction = allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
+        if (TransactionUtils.hasMissingSmartscanFields(transaction ?? null) && !ReportUtils.isSettled(transaction?.reportID)) {
+            return undefined;
+        }
+    } else if ((ReportUtils.isIOUReport(report) || ReportUtils.isExpenseReport(report)) && report?.ownerAccountID === currentUserAccountID) {
+        if (ReportUtils.shouldShowRBRForMissingSmartscanFields(report?.reportID ?? '-1') && !ReportUtils.isSettled(report?.reportID)) {
+            // Based on ReportUtils.shouldShowRBRForMissingSmartscanFields
+            return Object.values(reportActions ?? {}).find((action) => {
+                if (!ReportActionsUtils.isMoneyRequestAction(action)) {
+                    return false;
+                }
+                let transactionID = '-1';
+                if (ReportActionsUtils.isMoneyRequestAction(action)) {
+                    transactionID = ReportActionsUtils.getOriginalMessage(action)?.IOUTransactionID ?? '-1';
+                }
+                const transaction = allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
+                if (isEmptyObject(transaction)) {
+                    return false;
+                }
+                if (!ReportActionsUtils.wasActionTakenByCurrentUser(action)) {
+                    return false;
+                }
+                return TransactionUtils.hasMissingSmartscanFields(transaction);
+            });
```

Comment:
Cant we use `shouldShowRBRForMissingSmartscanFields` here? Because in case any changes happens there this logic will be outdated